src/BlockInfo.h

```diff
@@ -48,6 +48,10 @@ class cBlockInfo
 	inline static bool IsSolid                    (BLOCKTYPE a_Type) { return Get(a_Type).m_IsSolid;             }
 	inline static bool IsUseableBySpectator       (BLOCKTYPE a_Type) { return Get(a_Type).m_UseableBySpectator;  }
 	inline static bool FullyOccupiesVoxel         (BLOCKTYPE a_Type) { return Get(a_Type).m_FullyOccupiesVoxel;  }
+	inline static bool IsFullSolidOpaqueBlock     (BLOCKTYPE a_Type)
+	{
+		return (FullyOccupiesVoxel(a_Type) && IsSolid(a_Type) && (!IsTransparent(a_Type)));
```

Comment:
Which additional cases does this cover? I'm not entirely sure it is needed.

Reply:
https://minecraft.gamepedia.com/Lever
as described in minecraft wiki, lots of block such as lever required to be place on "full solid opaque block".